modules/payment/commerce_payment.post_update.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * @file
+ * Post update functions for Payment.
+ */
+
+/**
+ * Update user Remote ID field data using payments gateways instead of module.
+ *
+ * @see https://www.drupal.org/node/2861181
+ */
+function commerce_payment_post_update_1(&$sandbox) {
+  // Get the user ids that have 'commerce_remote_id' field not empty.
+  // First check is any user has this field set.
+  if (\Drupal::entityQuery('user')->exists('commerce_remote_id')->count()->execute() == 0) {
+    return t('No user Remote ID field data to update.');
+  }
+  $result = \Drupal::entityQuery('user')->exists('commerce_remote_id')->execute();
```

Comment:
We should run the count query only under if (!isset($sandbox['current'])) {
After that it's cached in $sandbox.

Then run the real query with a range each time.